### PR TITLE
Use TemplateSTRTree for pair finding in tree, per @dbaston

### PR DESCRIPTION
**Pull request by pramsey**

### CODE REVIEW

File: src/operation/relateng/EdgeSetIntersector.cpp [hunk 1]
```diff
@@ -68,26 +69,19 @@ EdgeSetIntersector::process(EdgeSegmentIntersector& intersector)
 {
     EdgeSegmentOverlapAction overlapAction(intersector);
 
-    for (const MonotoneChain& queryChain : monoChains) {
+    // Replaces JTS implementation that manaually iterates on the
```
Comment (rouault) [inline src/operation/relateng/EdgeSetIntersector.cpp:72]:
```suggestion
    // Replaces JTS implementation that manually iterates on the
```

**Comment by dbaston** [comment]
> This appears to have improved performance on my p-i-p test quite substantially (90s => 68s)

How does this compare to the old PreparedGeometry implementation?

**Comment by pramsey** [comment]
Aha, oh wait actually I accidentally flipped my tree back to the old implementation which is why it's faster. Note that the old implementation is in place still, the new implementation only kicks in for predicates that were not supported before.

File: src/operation/relateng/EdgeSetIntersector.cpp [hunk 2]
```diff
@@ -68,26 +69,19 @@ EdgeSetIntersector::process(EdgeSegmentIntersector& intersector)
...
-            /**
-             * following test makes sure we only compare each pair of chains once
-             * and that we don't compare a chain to itself
-             */
-            if (testChain->getId() <= queryChain.getId())
-                continue;
+        testChain->computeOverlaps(queryChain, &overlapAction);
+
+        if (intersector.isDone())
+            return;
```
Comment (dbaston) [inline src/operation/relateng/EdgeSetIntersector.cpp:82]: I think you want to return `false` here, from the docstring of `queryPairs`.

Comment (dbaston) [reply src/operation/relateng/EdgeSetIntersector.cpp:82]: See `MCIndexNoder::intersectChains` for an example

**Pull request closed without merging**